path_rotate_root.go

```diff
@@ -0,0 +1,241 @@
+package azuresecrets
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/go-uuid"
+	"github.com/hashicorp/vault-plugin-secrets-azure/api"
+	"github.com/hashicorp/vault/sdk/framework"
+	"github.com/hashicorp/vault/sdk/logical"
+	"github.com/mitchellh/mapstructure"
+)
+
+func pathRotateRoot(b *azureSecretBackend) *framework.Path {
+	return &framework.Path{
+		Pattern: "rotate-root",
+		Operations: map[logical.Operation]framework.OperationHandler{
+			logical.UpdateOperation: &framework.PathOperation{
+				Callback:                    b.pathRotateRoot,
+				ForwardPerformanceSecondary: true,
+				ForwardPerformanceStandby:   true,
+			},
+		},
+
+		HelpSynopsis: "Attempt to rotate the root credentials used to communicate with Azure.",
+		HelpDescription: "This path will attempt to generate new root credentials for the user used to access and manipulate Azure.\n" +
+			"The new credentials will not be returned from this endpoint, nor the read config endpoint.",
+	}
+}
+
+func (b *azureSecretBackend) pathRotateRoot(ctx context.Context, req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	config, err := b.getConfig(ctx, req.Storage)
+	if err != nil {
+		return nil, err
+	}
+
+	expDur := config.RootPasswordExpiration
+	if expDur == 0 {
+		expDur = rootPasswordExpiration
+	}
+	expiration := time.Now().Add(expDur)
+
+	passCred, err := b.rotateRootCredentials(ctx, req.Storage, *config, expiration)
+	if err != nil {
+		return nil, err
+	}
+
+	resultData := map[string]interface{}{
+		"secret_id": *passCred.KeyID,
```

Comment:
If we don't want to return this date on rotate-root, maybe we can move it to the config read endpoint.